rust/agama-lib/src/software/http_client.rs

```diff
@@ -0,0 +1,63 @@
+use crate::software::model::SoftwareConfig;
+use crate::{base_http_client::BaseHTTPClient, error::ServiceError};
+use std::collections::HashMap;
+
+pub struct SoftwareHTTPClient {
+    client: BaseHTTPClient,
+}
+
+impl SoftwareHTTPClient {
+    pub fn new() -> Result<Self, ServiceError> {
+        Ok(Self {
+            client: BaseHTTPClient::new()?,
+        })
+    }
+
+    pub fn new_with_base(base: BaseHTTPClient) -> Self {
+        Self { client: base }
+    }
+
+    pub async fn get_config(&self) -> Result<SoftwareConfig, ServiceError> {
+        self.client.get("/software/config").await
+    }
+
+    pub async fn set_config(&self, config: &SoftwareConfig) -> Result<(), ServiceError> {
+        // FIXME: test how errors come out:
```

Comment:
well, it can be improved by decoding body of error response and creating error from it, if it is also service error. I plan to do it, but other higher priorities come in the way.

Reply:
That's what I thought too, but 
1. in the end the code seeing these errors does no decisions, just shows them to the user
2. **if** we wanted to do some decisions, we would have to serialize them better than with `Display`/`Debug`